diff --git a/libft/include/env.h b/libft/include/env.h
index ef4b875..d80b429 100644
--- a/libft/include/env.h
+++ b/libft/include/env.h
@@ -3,10 +3,10 @@
 /*                                                        :::      ::::::::   */
 /*   env.h                                              :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: ebini <ebini@student.42lyon.fr>            +#+  +:+       +#+        */
+/*   By: ethebaul <ethebaul@student.42.fr>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2025/04/11 13:35:39 by ebini             #+#    #+#             */
-/*   Updated: 2025/06/15 01:01:29 by ebini            ###   ########lyon.fr   */
+/*   Updated: 2025/06/30 18:55:11 by ethebaul         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
@@ -57,4 +57,4 @@ int		ft_initenv(char **envp);
 int		ft_setenv(char *var_name, char *value);
 void	ft_unsetenv(char *var_name);
 
-#endif
\ No newline at end of file
+#endif
diff --git a/libft/src/env/env_unset.c b/libft/src/env/env_unset.c
index fbf1552..057c6b3 100644
--- a/libft/src/env/env_unset.c
+++ b/libft/src/env/env_unset.c
@@ -3,16 +3,18 @@
 /*                                                        :::      ::::::::   */
 /*   env_unset.c                                        :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: ebini <ebini@student.42lyon.fr>            +#+  +:+       +#+        */
+/*   By: ethebaul <ethebaul@student.42.fr>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2025/04/30 23:25:13 by ebini             #+#    #+#             */
-/*   Updated: 2025/04/30 23:52:05 by ebini            ###   ########lyon.fr   */
+/*   Updated: 2025/06/30 19:05:43 by ethebaul         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "env.h"
 #include "libft.h"
 
+#include <stdio.h>
+
 void	env_unset(t_env *env, char *var_name)
 {
 	const size_t	name_len = ft_strlen(var_name);
@@ -21,8 +23,8 @@ void	env_unset(t_env *env, char *var_name)
 	env_tab = env->data;
 	while (*env_tab)
 	{
-		if (starts_by(*env_tab, var_name) && *env_tab[name_len] == '=')
-			*env_tab[name_len] = '\0';
+		if (starts_by(*env_tab, var_name) && (*env_tab)[name_len] == '=')
+			(*env_tab)[name_len] = '\0';
 		++env_tab;
 	}
 }
